ls1intum/Artemis
# `Development`: Increase code quality for mail service and notification system

### Checklist
#### General
#### Server

### Motivation and Context
This is a small follow-up PR for #4303.
I implemented the change requests by @Olimbia23 that increase the code quality, especially of the MailService.
This PR also adds a new ExerciseType enum and functionality so you can now instead of typecasting an exercise to check what subtype it has you can simply call the new getExerciseType method.

### Description
I mostly refactored a few parts of the MailService code and extracted some of its logic elsewhere where it can be reused for others.

The changes in the notification services are simple: The switch cases had redundant/duplicate information. I combined similar cases together.

Most changed LoC are due to a simple static import of enums, i.e. instead of a redundant and repeating `NotificationType.EXERCISE_RELEASED` -> simply `EXERCISE_RELEASED`

### Steps for Testing
Look at the code.
Try to generate a few emails (especially exercise-released emails) and check if they are still valid.

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
I have not touched the logic. This is a very minor refactoring.
